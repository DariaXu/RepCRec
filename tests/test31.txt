// Test that sites that are down on end() will cause failure for non read-only
// transactions.
begin(T1)
beginRO(T2)
R(T2, x1)
W(T1, x1, 81)
begin(T3)
R(T3, x3)
begin(T4)
R(T4, x5); W(T4, x5, 9)
fail(2)
// should abort since 2 is down
end(T1) 
fail(4)
// should abort since 4 is down
end(T3) 
fail(6)
// should abort since 6 is down
end(T4) 
// should commit since read-only
end(T2) 
dump()
// ---
// assertAborted(T1)
// assertCommitted(T2)
// assertAborted(T3)
// assertAborted(T4)
