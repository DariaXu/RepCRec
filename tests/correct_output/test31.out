T2 reads on Site 2 - x1: 10
Site 2: T1 write x1=81
T3 reads on Site 4 - x3: 30
T4 reads on Site 6 - x5: 50
Site 6: T4 write x5=9
Abort: T1
Abort: T3
Abort: T4
Commit: T2
Site 1 - x2: 20, x4: 40, x6: 60, x8: 80, x10: 100, x12: 120, x14: 140, x16: 160, x18: 180, x20: 200
Site 2 - x1: 10, x2: 20, x4: 40, x6: 60, x8: 80, x10: 100, x11: 110, x12: 120, x14: 140, x16: 160, x18: 180, x20: 200
Site 3 - x2: 20, x4: 40, x6: 60, x8: 80, x10: 100, x12: 120, x14: 140, x16: 160, x18: 180, x20: 200
Site 4 - x2: 20, x3: 30, x4: 40, x6: 60, x8: 80, x10: 100, x12: 120, x13: 130, x14: 140, x16: 160, x18: 180, x20: 200
Site 5 - x2: 20, x4: 40, x6: 60, x8: 80, x10: 100, x12: 120, x14: 140, x16: 160, x18: 180, x20: 200
Site 6 - x2: 20, x4: 40, x5: 50, x6: 60, x8: 80, x10: 100, x12: 120, x14: 140, x15: 150, x16: 160, x18: 180, x20: 200
Site 7 - x2: 20, x4: 40, x6: 60, x8: 80, x10: 100, x12: 120, x14: 140, x16: 160, x18: 180, x20: 200
Site 8 - x2: 20, x4: 40, x6: 60, x7: 70, x8: 80, x10: 100, x12: 120, x14: 140, x16: 160, x17: 170, x18: 180, x20: 200
Site 9 - x2: 20, x4: 40, x6: 60, x8: 80, x10: 100, x12: 120, x14: 140, x16: 160, x18: 180, x20: 200
Site 10 - x2: 20, x4: 40, x6: 60, x8: 80, x9: 90, x10: 100, x12: 120, x14: 140, x16: 160, x18: 180, x19: 190, x20: 200
